pkg/router/controller/unique_host.go

```diff
@@ -131,6 +131,24 @@ func (p *UniqueHost) HandleRoute(eventType watch.EventType, route *routeapi.Rout
 				}
 			}
 			if !added {
+				// Clean out any old form of this route
+				next := []*routeapi.Route{}
+				for i := range old {
+					if old[i].Name != route.Name {
```

Comment:
We already checked the namespace before it can get in the list.  And the delete loop at the bottom does the same check I implemented.  However, if you want, I can add the namespace check too.